src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -213,4 +213,21 @@ private function hasAttributeAccessorMethod(string $class, string $attribute): b
             && !$method->getAttributes(Ignore::class)
             && !$method->getNumberOfRequiredParameters();
     }
+
+    private function isReadableCache(object|string  $class, object|string $classOrObject, string $attribute): bool
```

Comment:
Could we remove this function and keep the check inlined? It's on the hot path, an extra function call can have significant effects on performance.

Reply:
I have revert the check to inline. I have also adjusted the [Test](https://github.com/symfony/symfony/pull/58255#discussion_r1780058718).